Content/NPCs/Town/MorvenNPC.cs

```diff
@@ -251,6 +253,30 @@ private void PathedMovement()
 		}
 	}
 
+	private void RavenStatueNearby(ref Vector2 target, out bool statue)
+	{
+		const int Size = 40;
+
+		Point16 pos = NPC.Center.ToTileCoordinates16();
+
+		for (int i = pos.X - Size; i < pos.X + Size; ++i)
+		{
+			for (int j = pos.Y - Size; j < pos.Y + Size; ++j)
+			{
+				Tile tile = Main.tile[i, j];
+
+				if (tile.HasTile && tile.TileType == ModContent.TileType<RavenStatue>())
+				{
+					target = new Vector2(i, j) * 16;
```

Comment:
IOOB guaranteed. Also this would be better off being cached across 5-10 ticks.
```suggestion
		const int Size = 40;

		Point16 pos = NPC.Center.ToTileCoordinates16();
		(int xStart, int xEnd) = (Math.Max(0, pos.X - Size), Math.Min(pos.X + Size, Main.maxTilesX));
		(int yStart, int yEnd) = (Math.Max(0, pos.Y - Size), Math.Min(pos.Y + Size, Main.maxTilesY));

		for (int x = xStart; x < xEnd; x++)
		{
			for (int y = yStart; y < yEnd; y++)
			{
				Tile tile = Main.tile[x, y];

				if (tile.HasTile && tile.TileType == ModContent.TileType<RavenStatue>())
				{
					target = new Vector2(x, y) * 16;
```